core/state_processor_rip7560.go

```diff
@@ -211,8 +176,19 @@ func ApplyRip7560ValidationPhases(chainConfig *params.ChainConfig, bc ChainConte
 		GasPrice: gasPrice,
 	}
 	evm := vm.NewEVM(blockContext, txContext, statedb, chainConfig, cfg)
+	epc := &EntryPointCall{}
+
+	if evm.Config.Tracer == nil {
+		evm.Config.Tracer = &tracing.Hooks{
+			OnEnter: epc.OnEnter,
+		}
+	} else {
+		// keep the original tracer's OnEnter hook
+		epc.OnEnterSuper = evm.Config.Tracer.OnEnter
+		evm.Config.Tracer.OnEnter = epc.OnEnter
```

Comment:
This is a change to an input parameter.
You assume your caller created the evm.Config.Tracer just for this call.
At a minimum - duplicate the Tracer before modifying it.

Reply:
(unfortunately, it is impossible to copy it locally: it requires copying entire "evm" struct, and inside it entire Config struct, and then Tracer struct just to modify a single field...